src/plugins/dashboard/public/application/hooks/use_dashboard_app_state.ts

```diff
@@ -148,6 +150,15 @@ export const useDashboardAppState = ({
       if (canceled || !loadSavedDashboardResult) return;
       const { savedDashboard, savedDashboardState } = loadSavedDashboardResult;
 
+      // If the saved dashboard is an alias match, then we will redirect
+      if (savedDashboard.outcome === 'aliasMatch') {
+        if (screenshotModeService?.isScreenshotMode()) {
+          scopedHistory().replace(savedDashboard.getFullPath());
+        } else {
+          spacesService?.ui.redirectLegacyUrl(savedDashboard.getFullPath());
```

Comment:
This uses `navigateToApp` with the current `appId` under the hood.
That means that when you go to this URL:
```
/s/test2/app/dashboards#/view/7c0a5590-10e5-11ec-8e89-e5edd366fc1d-test2
```
Kibana detects the aliasMatch outcome and redirects you here:
```
/s/test2/app/dashboards/app/dashboards#/view/7c0a5590-10e5-11ec-8e89-e5edd366fc1d-test2-newId
```

So you're actually at the `/app/dashboards/app/dashboards` page, which actually loads because of a quirk in our router, but we should change it so we don't have the app show up in the path twice 😅 
Perhaps you can expose a `getLocalPath()` method?